analyzers/hardcodedNonce.go

```diff
@@ -0,0 +1,231 @@
+	// It goes and check if this function contains call to crypto/rand.Read in it's body(Assuming that calling crypto/rand.Read in a function, is used for the generation of nonce/iv )
+	case *ssa.Call:
+		if calledFunction, ok := valType.Call.Value.(*ssa.Function); ok {
+			if contains, funcErr := isFuncContainsCryptoRand(calledFunction); !contains && funcErr == nil {
+				issueDescription += " by passing a value from function which doesn't use crypto/rand"
+				tmp, hasErr := iterateThroughReferrers(val, funcsToTrack, pass.Analyzer.Name, issueDescription, pass.Fset, issue.High, issue.Medium)
+				gosecIssue = append(gosecIssue, tmp...)
+				err = hasErr
+			} else if funcErr != nil {
+				err = funcErr
+			}
+		}
+
+	// only checks from strings->[]byte
+	// might need to add additional types
+	case *ssa.Convert:
+		if valType.Type().String() == "[]byte" && valType.X.Type().String() == "string" {
+			issueDescription += " by passing converted string"
+			tmp, hasErr := iterateThroughReferrers(val, funcsToTrack, pass.Analyzer.Name, issueDescription, pass.Fset, issue.High, issue.High)
+			gosecIssue = append(gosecIssue, tmp...)
+			err = hasErr
+		}
+
+	case *ssa.Parameter:
+		//arg given to tracked function is wrapped in another function, example:
+		// func encrypt(..,nonce,...){
+		// aesgcm.Seal()
+		//}
+		// save parameter position, by checking the name of the variable used in tracked functions and comparing it with the name of the arg
+		if valType.Parent() != nil {
+			trackedFunctions := make(map[string][]int)
+			for index, funcArgs := range valType.Parent().Params {
+				if funcArgs.Name() == valType.Name() && funcArgs.Type() == valType.Type() {
+					trackedFunctions[valType.Parent().String()] = []int{len(valType.Parent().Params), index}
+				}
+			}
+			result := iterateAndGetArgsFromTrackedFunctions(ssaFuncs, &trackedFunctions)
+
+			issueDescription += " by passing a parameter to a function and"
+			// recursively backtrack to where the origin of a variable passed to multiple functions is
+			for _, trackedVariable := range *result {
+
+				tmp, hasErr := raiseIssue(trackedVariable, &trackedFunctions, ssaFuncs, pass, issueDescription)
+				gosecIssue = append(gosecIssue, tmp...)
+				err = hasErr
+			}
+		}
+	}
+	return gosecIssue, err
+}
+
+// Iterate through all places that use the `variable` argument and check if it's used in one of the tracked functions
+func iterateThroughReferrers(variable *ssa.Value, funcsToTrack *map[string][]int, analyzerID string, issueDescription string, fileSet *token.FileSet, issueSeverity issue.Score, issueConfidence issue.Score) ([]*issue.Issue, error) {
+	if funcsToTrack == nil || variable == nil || analyzerID == "" || issueDescription == "" || fileSet == nil {
+		return nil, errors.New("received a nil object")
+	}
+	var gosecIssues []*issue.Issue = nil
+	// Go trough all functions that use the given arg variable
+	for _, referrer := range *(*variable).Referrers() {
```

Comment:
I think, you don't need to dereference the `variable` pointer, the compiler does that automatically when calling the Referrers method.